definitions/npm/cypress_v5.x.x/flow_v0.25.x-/cypress_v5.x.x.js

```diff
@@ -0,0 +1,803 @@
+   */
+  viewport(
+    preset: Cypress$ViewportPreset,
+    orientation?: Cypress$ViewportOrientation,
+    options?: Cypress$Loggable,
+  ): void,
+  viewport(
+    width: number,
+    height: number,
+    options?: Cypress$Loggable,
+  ): void,
+
+  /**
+   * @see https://docs.cypress.io/api/commands/visit.html
+   */
+  visit(url: string, options?: Cypress$VisitOptions): Cypress$Window,
+  visit(options: Cypress$VisitOptions): Cypress$Window,
+
+  /**
+   * @see https://docs.cypress.io/api/commands/wait.html
+   */
+  wait(ms: number, options?: Cypress$LoggableTimeoutable): Cypress$Global,
+  wait(alias: string, options?: Cypress$LoggableTimeoutable): any,
+  wait(aliases: Array<string>, options?: Cypress$LoggableTimeoutable): any,
+
+  /**@see https://docs.cypress.io/api/commands/window.html
+   */
+  window(options?: Cypress$Loggable): Cypress$Window,
+
+  /**
+   * @see https://docs.cypress.io/api/commands/within.html
+   */
+  within(fn: Function): Cypress$Global,
+  within(options: Cypress$Loggable, fn: Function): Cypress$Global,
+
+  /**
+   * @see https://docs.cypress.io/api/commands/wrap.html
+   */
+  wrap(object: { ... }, options?: Cypress$Loggable): { [key: string]: any, ... },
+
+  /**
+   * @see https://docs.cypress.io/api/commands/writefile.html
+   */
+  writeFile(
+    filePath: string,
+    contents: Cypress$FileContents,
+    options?: Cypress$Timeoutable,
+  ): Cypress$FileContents,
+  writeFile(
+    filePath: string,
+    contents: Cypress$FileContents,
+    encoding: Cypress$Encoding,
+    options?: Cypress$Timeoutable,
+  ): Cypress$FileContents,
+}
+
+declare interface Cypress$DebugOptions {
+  verbose?: boolean
+}
+
+declare interface Cypress$Loggable {
+  log?: boolean
+}
+
+declare interface Cypress$Timeoutable {
+  timeout?: number
+}
+
+declare type Cypress$LoggableTimeoutable = { ... } & Cypress$Loggable & Cypress$Timeoutable
+
+declare type Cypress$ClockObject = {|
+  tick: (milliseconds: number) => Promise<Cypress$ClockObject>,
+  restore: () => void,
+|};
+
+declare type Cypress$BlurOptions = {|
+  force?: boolean
+|} & Cypress$Loggable & Cypress$Timeoutable
+
+declare type Cypress$CheckOptions = {|
+  interval?: number,
+  force?: boolean
+|} & Cypress$Loggable & Cypress$Timeoutable
+
+declare type Cypress$ClearOptions = {|
+  force?: boolean,
+  interval?: number
+|} & Cypress$Loggable & Cypress$Timeoutable
+
+declare type Cypress$ClickOptions = {|
+  force?: boolean,
+  multiple?: boolean,
+  interval?: number
+|} & Cypress$Loggable & Cypress$Timeoutable
+
+declare type Cypress$ExecOptions = {|
+  failOnNonZeroExit?: boolean,
+  env?: Object
+|} & Cypress$Loggable & Cypress$Timeoutable
+
+declare type Cypress$RequestOptions = {|
+  auth?: Object,
+  body?: Cypress$RequestBody,
+  failOnStatusCode?: boolean,
+  followRedirect?: boolean,
+  form?: boolean,
+  gzip?: boolean,
+  headers?: Object,
+  method?: Cypress$HttpMethod,
+  qs?: string,
+  url: string
+|} & Cypress$Loggable & Cypress$Timeoutable
```

Comment:
**suggestion:** prefer spread to intersection for exact objects.

Reply:
Do you mean to say, change it to this instead?

For some reason when I do this, method overloading doesn't work anymore for the functions that use this type as an option, any idea how I could fix this?
```
declare type Cypress$RequestOptions = {|
  ...Cypress$Loggable,
  ...Cypress$Timeoutable,
  auth?: Object,
  body?: Cypress$RequestBody,
  failOnStatusCode?: boolean,
  followRedirect?: boolean,
  form?: boolean,
  gzip?: boolean,
  headers?: Object,
  method?: Cypress$HttpMethod,
  qs?: string,
  url: string
|}
```